pc/PlatformDeviceControl.c

```diff
@@ -157,27 +113,64 @@ int XLinkPlatformBootRemote(deviceDesc_t* deviceDesc, const char* binaryPath)
     rewind(file);
     if(file_size <= 0 || !(image_buffer = (char*)malloc(file_size)))
     {
-        if(usb_loglevel)
-            perror("buffer");
+        mvLog(MVLOG_ERROR, "cannot allocate image_buffer. file_size = %ld", file_size);
         fclose(file);
         return -3;
     }
     if(fread(image_buffer, 1, file_size, file) != file_size)
     {
-        if(usb_loglevel)
-            perror(binaryPath);
+        mvLog(MVLOG_ERROR, "cannot read file to image_buffer");
         fclose(file);
         free(image_buffer);
         return -7;
     }
     fclose(file);
 
-    rc = XLinkPlatformBootMemoryRemote(deviceDesc, image_buffer, file_size);
+    if(XLinkPlatformBootFirmware(deviceDesc, image_buffer, file_size)) {
+        free(image_buffer);
+        return -1;
+    }
+
     free(image_buffer);
+    return 0;
+}
 
-    return rc;
+int XLinkPlatformBootFirmware(const deviceDesc_t* deviceDesc, const char* firmware, size_t length) {
+    if (deviceDesc->protocol == X_LINK_PCIE) {
+        // Temporary open fd to boot device and then close it
+        int* pcieFd = NULL;
+        int rc = pcie_init(deviceDesc->name, (void**)&pcieFd);
+        if (rc) {
+            return rc;
+        }
+#if (!defined(_WIN32) && !defined(_WIN64))
+        rc = pcie_boot_device(*(int*)pcieFd, firmware, length);
+#else
+        rc = pcie_boot_device(pcieFd, firmware, length);
+#endif
+        pcie_close(pcieFd); // Will not check result for now
+        return rc;
+    } else if (deviceDesc->protocol == X_LINK_USB_VSC) {
+
+        char subaddr[28+2];
```

Comment:
use a `const int` variable for size?

Reply:
This is an upstream change (same as before actually just the function moved around) - not exactly sure why 28. I'd leave as is for now